### PR TITLE
Adjust payment-dummy-driver to new api

### PR DESCRIPTION
**Pull request by maaktweluit**
work in progress...

- moved `payment-driver/dummy.rs` to its own crate: `payment-driver/dummy/lib.rs`
- removed all GNT-driver related files
- update feature toggle to be on the import of `PaymentDriverService`
- Code is compiling, not working

### CODE REVIEW

File: core/payment-driver/dummy/src/service.rs [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
 use ya_persistence::executor::DbExecutor;
 use ya_service_bus::{typed as bus, RpcEndpoint};
 
+
+const BUS_ID_POSTFIX: &'static str = "dummy";
+
+
 pub fn bind_service(db: &DbExecutor, processor: PaymentDriverProcessor) {
     log::debug!("Binding payment driver service to service bus");
+    let bus_id: &str = &(BUS_ID_PREFIX.to_owned() + BUS_ID_POSTFIX);
```
Comment (maaktweluit) [inline core/payment-driver/dummy/src/service.rs:13]: why are strings in rust so hard..

would like: ```
const BUS_ID: &'static str = BUS_ID_PREFIX + "dummy";
```

File: core/payment-driver/dummy/src/service.rs [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
 use bigdecimal::BigDecimal;
 use ya_core_model::driver::*;
 use ya_persistence::executor::DbExecutor;
 use ya_service_bus::{typed as bus, RpcEndpoint};
 
+
+const BUS_ID_POSTFIX: &'static str = "dummy";
+
+
 pub fn bind_service(db: &DbExecutor, processor: PaymentDriverProcessor) {
```
Comment (maaktweluit) [inline core/payment-driver/dummy/src/service.rs:11]: preferably we move this back to `payment-driver` and pass the full bus_id

File: core/payment-driver/dummy/src/lib.rs [hunk 3]
```diff
@@ -54,65 +57,40 @@ impl PaymentDriver for DummyDriver {
...
         let payments = self.payments.clone();
         let details = PaymentDetails {
             recipient: recipient.to_string(),
             sender: sender.to_string(),
-            amount: amount.base_currency_amount,
+            amount: amount,
             date: Some(Utc::now()),
         };
-        let invoice_id = invoice_id.to_string();
+        // FIXME: generate payment_order_id?
```
Comment (maaktweluit) [inline core/payment-driver/dummy/src/lib.rs:79]: 2 questions on payment_order_id:
- Where should it be generated? just a new UUID?
- Should it be part of `PaymentDetails`?

File: core/payment-driver/dummy/src/lib.rs [hunk 4]
```diff
@@ -54,65 +57,40 @@ impl PaymentDriver for DummyDriver {
...
+        let payment_order_id = "123".to_string();
 
         Box::pin(async move {
-            match payments.lock().await.entry(invoice_id.clone()) {
-                Entry::Occupied(_) => Err(PaymentDriverError::PaymentAlreadyScheduled(invoice_id)),
+            match payments.lock().await.entry(payment_order_id.clone()) {
+                // FIXME: need other properties to check if Payment is already Scheduled
```
Comment (maaktweluit) [inline core/payment-driver/dummy/src/lib.rs:84]: Depending on the question above, how can we ensure payments are only scheduled once? Or is this no longer the driver repsonsibility

File: core/payment-driver/dummy/src/lib.rs [hunk 5]
```diff
@@ -126,11 +104,21 @@ impl PaymentDriver for DummyDriver {
...
-            currency: Currency::Gnt,
-            amount: utils::str_to_big_dec(&amount).unwrap(),
-        })))
+        Box::pin(future::ready(Ok(utils::str_to_big_dec(&amount).unwrap())))
+    }
+}
+
+pub struct PaymentDriverService;
+
+impl PaymentDriverService {
```
Comment (maaktweluit) [inline core/payment-driver/dummy/src/lib.rs:115]: could be moved back to `payment-driver` if we pass the ::new() function to use

File: core/serv/src/main.rs [hunk 6]
```diff
@@ -24,6 +24,10 @@ use ya_activity::service::Activity as ActivityService;
 use ya_identity::service::Identity as IdentityService;
 use ya_net::Net as NetService;
 use ya_payment::PaymentService;
+
+#[cfg(feature = "dummy-driver")]
```
Comment (maaktweluit) [inline core/serv/src/main.rs:28]: This approach makes it only possible to import one driver at the time, think we need to fix that enum to import both.

File: core/payment-driver/dummy/src/service.rs (hunk 1 above)
Comment (Wiezzel) [reply core/payment-driver/dummy/src/service.rs:13]: You can use `lazy_static` to have it as `&'static str`.

File: core/payment-driver/dummy/src/lib.rs (hunk 3 above)
Comment (Wiezzel) [reply core/payment-driver/dummy/src/lib.rs:79]: Just a new UUID. Even a counter incremented with every new order should do.

Nope, it doesn't need to go to payment details.

File: core/payment-driver/dummy/src/lib.rs (hunk 4 above)
Comment (Wiezzel) [reply core/payment-driver/dummy/src/lib.rs:84]: Not the driver's responsibility. Twice ordered, twice served.

File: core/serv/src/main.rs (hunk 6 above)
Comment (Wiezzel) [reply core/serv/src/main.rs:28]: I'm working on it but it's difficult.

File: core/payment-driver/dummy/src/lib.rs (hunk 3 above)
Comment (maaktweluit) [reply core/payment-driver/dummy/src/lib.rs:79]: updated it to be an incremental counter

File: core/payment-driver/src/dao/payment.rs [hunk 7]
```diff
@@ -48,49 +48,49 @@ impl<'c> PaymentDao<'c> {
...
-                *confirmation = PaymentConfirmation {
-                    confirmation: tx_hash,
-                };
-            }
-
-            Ok(Some(status))
-        })
-        .await
-    }
+    // pub async fn get_payment_status(&self, invoice_id: String) -> DbResult<Option<PaymentStatus>> {
```
Comment (maaktweluit) [inline core/payment-driver/src/dao/payment.rs:51]: @bidzyyys these break my compilation, please bring them back ( or delete them ) when updating the GNT driver

File: core/serv/src/main.rs (hunk 6 above)
Comment (Wiezzel) [reply core/serv/src/main.rs:28]: Not really worth the pain, let's leave it like this.

File: core/payment-driver/src/dao/payment.rs (hunk 7 above)
Comment (maaktweluit) [reply core/payment-driver/src/dao/payment.rs:51]: yes, when these functions are commented out it compiles :)

Thank you <3

**Pull request merged**